## Step 1: A Framework for Unlocking and Linking WWII Japanese American Incarceration Biographical Data - Context Based Data Manipuation and Analysis Process- Part 1 

Data manipuation is the process of fine tuning data to organize and prepare the data for research purposes. Data manipulation can include sorting, filtering, cleaning, normalizing, and joining separate datasets. Once the data has been prepared, data analysis can be perfomed. According to the Office of Research Integrity, "Data Analysis is the process of systematically applying statistical and/or logical techniques to describe and illustrate, condense and recap, and evaluate data." Below are the steps performed for data manipulation and analysis process using Python programming language on the Japanese American World War II datasets. The modules are split into two parts - Part 1 and Part 2.

### Acquiring or Accessing the Data

The data for this project comes from three disparate collections housed at the National Archives and Records Administration (NARA) this includes the War Relocation Authority (WRA), Final Accountability Roster (FAR), and Internal Security Case Reports, also referred to as Index Card or Incident Cards. The data sources are included in this module for the WRA and FAR as comma separated values (csv) files. The link below will direct you to those data files: 

WRA [link]
FAR [link]
Incident Card upon request

The WRA dataset has X rows of data, the FAR has X rows, and the Incident Card has X rows.

To work with a csv file in Python, one of the first steps is to import a Python library called as 'pandas' which will help the program convert the csv file into a dataframe format or commonly called as a table format. We import the library into the program as shown below:

In [1]:
# Import libraries
# Pandas is used for data science/data analysis and machine learning tasks 
import pandas as pd
# Numpy provides support for multi-dimensional arrays
import numpy as np

Using the pandas library, new dataframes can be created from the datasets imported into the notebook. Once the dataframes have been created from the imported files, the print() function is used to display the top first 5 rows in the dataframe.

In [2]:
# Creating a dataframe is essentially creating a table-like data structure that can read csv files, flat files, and other delimited data.
# Converting input data into a dataframe is the starting point with Python programming language for big data analytics
# Below command reads in the WRA, FAR, and Incident card datasets. The WRA and FAR should already be loaded in a folder called 'Datasets' wra_cleanup-09-26-2021.csv and Copy-far_cleanup-09-26-2021.csv
wradf = pd.read_csv('wra_cleanup-09-26-2021.csv',dtype=object,na_values=[],keep_default_na=False)
fardf = pd.read_csv('far_cleanup-09-26-2021.csv',dtype=object,na_values=[],keep_default_na=False)
incidentdf = pd.read_csv('incidentdfv1.csv',dtype=object,na_values=[],keep_default_na=False)

# Below command prints the first 5 records of the Incident Cards after the data is copied from the csv file
incidentdf.head(5)

FileNotFoundError: [Errno 2] No such file or directory: 'far_cleanup-09-26-2021.csv'

Below are the variables in the Incident Cards that were chosen to be cleaned and manipulated for use in the following steps.

Offense -- This indicates the offenses assigned to Japanese Americans by government officials
Case # -- A case number was assigned according to the type of offense
Date -- This indicates the date the offense took place 
Residence -- The location of where that specific individual resided 
Other -- This indicates additional notes that were recorded by officials on the Index, "Incident Cards"
First Name -- The first name of the incarcerated Japanese American
Last Name -- The last name of the incarcerated Japanese American


Below are the variables in the WRA and FAR that were chosen to be cleaned and manipulated for use in the following steps.
First Name -- The first name of the incarcerated Japanese American
Last Name -- The last name of the incarcerated Japanese American
Family Number -- A number assigned to families upon entry into the assembly centers


We anticipated errors and misinterpretation of names, numbers, etc. in the data since these databases were either transcribed manually or physical copies digitized using OCR techniques. We approached this by individually exploring and cleaning the selected columns using the text and numerical operation functions in Python. To begin, we looked at the datasets holistically to identify variables that would allow for us to generate meaningful stories and visualizations. After a list of variables were confirmed, we analyzed each of them in detail to document bad data and eliminate them if possible, modify data types, exclude them from the final visualizations if found to be invalid, etc.

This project involved team members with diverse backgrounds and knowledge in the technical, historical, and archival domains. The project presented opportunties for collaboration and individual work. We decided it was more productive to have a hybrid approach where we analyzed the data on our own and created sub-projects and reported our results and progress back to the group for discussion. With respect to the analysis performed on the dataset, decisions were data-driven or historical facts driven.

Through researching the literature, conversations with community partner Geoff Froh, from Densho, and experts in the field, the team members were able to follow steps where specific variables were identified and shared among the entire group for their inputs before finalizing the results. 

## Case Number (Issue of Incident Card)
Through many conversations and pulling in our differing expertise we were able to develop ideas that let us integrate a range of methods and techniques to exhibit multifaceted perspectives of incarcerated Japanese Americans during World War II. Out of these discussions one of the many sub projects that developed was Spatial Analysis of incarcerated Japanese Americans recorded as being invovled in the uprising ('riot') at Tule Lake on November 4, 1943. 

Additionally, much of our discussion was also centered on how the data should be analyzed, presented, and shared without impacting the sensitivity of the people involved, especially since this set of collection is unique.

One of the variables conisdered for this project was case number. This field indicates the case number assigned to offenses that occured at Tule Lake. Due to the case number being entered on the index cards by different government personnel the formatting was inconsistent. There were a number of issues with the case number field in the original dataset. Examples of the variations of case number formats includes 'A7-'; ‘A-7 ’; A-7- also have 'A-7-P' also have 'A-7-p'; 'A-7p' Also have 'A-7P'; ‘A-7,’; 'A7,'. 

In [ ]:
# The below command prints out the descriptive details of the column 'casenumber'
incidentdf["casenumber"].describe()

count     15748
unique     4475
top            
freq        680
Name: casenumber, dtype: object

: 

: 

In [ ]:
# The below command displays an array of unique date values in the 'casenumber' column
incidentdf["casenumber"].unique()

array(['A-607', 'A7-P138', 'A-7 Page 1465', ..., '#477', 'A-7 Page 1393',
       'A-7-P180'], dtype=object)

: 

: 

### Indentifying Cases of 'Riot'

As can be seen from above, there are varying formats for case number, some have dashes, others have a lowercase or a capital 'P', and some are missing the letter but have the word 'Page', etc. To identify and capture all of the variations of case number we can use regular expressions within the contains() function as shown below: 

In [ ]:
# Apply regex will help us find all those variations
caseno = incidentdf[incidentdf['casenumber'].str.contains("A[-\s]?7([p,?-p\s-].*)?$", na=False)]

/opt/conda/lib/python3.8/site-packages/pandas/core/strings/accessor.py:101: UserWarning: This pattern has match groups. To actually get the groups, use str.extract.
  return func(self, *args, **kwargs)


: 

: 

In [ ]:
# The below command prints out the number of times these variations appear within the dataframe. 
caseno['casenumber'].unique

<bound method Series.unique of 1              A7-P138
2        A-7 Page 1465
14            A-7-P284
20        A-7 Page 543
26            A-7-P284
             ...      
15742    A-7 Page 1380
15743    A-7 Page 1376
15744    A-7 Page 1377
15745         A-7-P180
15746          A-7-P95
Name: casenumber, Length: 3850, dtype: object>

: 

: 

As can be seen above there are, give or take, about 3,850 cases associated with 'riot' that meet the regular expression criteria used to capture several variations. 

**Note, this number does not necesarily mean we captured all of the case numbers linked to 'riot',. Since the data was manually entered on the index cards by government personnel it is likely there are errors and some cards tied to 'riot' are not captured in this variation. 

Another alternative for looking for cards linked to 'riot' is to search through the offense column. As shown below. 

In [ ]:
# Search through the 'newoffense' column using contains function
noffriot = incidentdf[incidentdf['newoffense'].str.contains('riot')]
noffriot.count()

Image#                    3872
newdate                   3872
casenumber                3872
offense                   3872
newoffense                3872
DEATH                     3872
name                      3872
lastname                  3872
firstname                 3872
Other Names (known as)    3872
Job/Role/Title            3872
Residence                 3872
NEW-residence             3872
X                         3872
Y                         3872
otherpeople               3872
Age                       3872
Family / Alien No.        3872
Other                     3872
Unnamed: 19               3872
dtype: int64

: 

: 

After performing the filter, the output of cards linked to 'riot' is higher than that of the case number column. Let's apply the unique function to view what unique values the contains function captured when 'riot' was searched for in the new offense column. 

In [ ]:
# the unique function will return unique values in a series aka column. 
noffriot['newoffense'].unique()

array(['riot'], dtype=object)

: 

: 

As shown above other values that contain the word 'riot' like 'disorderly conduct false riot call' and 'assualt attempted riot' are also returned. Though these cases may be associated with the 'riot' that occured on 11/4/1944 we want to focus on cases that only contain 'riot'. 

Regular expressions can also be applied here to filter and only return values that contain 'riot'. Regular expressions are a sequence of characters, that can sometimes comprise of several characters, used to search through strings for patterns. Though regular expressions are helpful and a powerful tool, they can also be challenging as some characters have to be in a specific order. Here are a few helpful tools to introduce regular expressions: [Regular Expressions HOWTO](https://docs.python.org/3/howto/regex.html), [pythex.org](https://pythex.org), [How to use Regex in Pandas](https://kanoki.org/2019/11/12/how-to-use-regex-in-pandas/), [AutoRegex](https://www.autoregex.xyz).  

Below we use the ^ regular expression to return values where 'riot' is at the start of the string. By simply using this character we have reduced the number of values returned which more closely aligns with the number of cases linked to 'riot'.  

In [ ]:
# The ^ is used to search for values at the start of a string. 
noffriot = incidentdf[incidentdf['newoffense'].str.contains('^riot')]
noffriot.count()

Image#                    3860
newdate                   3860
casenumber                3860
offense                   3860
newoffense                3860
DEATH                     3860
name                      3860
lastname                  3860
firstname                 3860
Other Names (known as)    3860
Job/Role/Title            3860
Residence                 3860
NEW-residence             3860
X                         3860
Y                         3860
otherpeople               3860
Age                       3860
Family / Alien No.        3860
Other                     3860
Unnamed: 19               3860
dtype: int64

: 

: 

To further explore the differences in the results of 'riot' in the new offense and the case number column we can use OpenRefine. OpenRefine will allow us to quickly view, filter, cluster and confirm the accuracy of case numbers associated with 'riot'. 

In the instance seen below, the difference is a result of the case number being inncorrectly entered or designated at the initial time of entry. **NOTE: It is unclear whether this error is an outcome of the initial data entry or a result of the data being manually entered from the index cards to a spreadsheet.  

<center><img src="OpenRefine-newoffense-data-1.png" width="800" alt="Locations"></center>
<center><img src="OpenRefine-newoffense-data-2.png" width="800" alt="Locations"></center>

More time could be spent refining the regular expression string used to search through the case numbers, but this is a good opportunity to practice using regular expressions on your own.

## Discovering Influential Figures in the 'Riot'

Once the incident card data has been fine tuned and reflects cases only related to 'riot' we can then begin exploring the dataset to discover individuals who's names appears the most among the 'riot' cards. The decision was made to begin this project by focusing on the key figures invovled in the 'riot' with the possiblility of expanding and applying these methods to a larger dataset. 

The case number series that was filtered using the contains function earlier in the process will be used to identify the individuals recorded as being part of the 'riot'. As a reminder, when the data was filtered, caseno was assigned to that new output so we know that we are working specifically with data associated with case numbers linked to 'riot'. By applying values count to the caseno series we can identify these key or influential figures. 

The values count function will return objects, in descending order, that contain counts of unique values. So the first element will always be the most frequently occuring element, as shown below: 

In [ ]:
# The below command returns count values of the names in the caseno series
riotcount = caseno['name'].value_counts()
riotcount.head(30)

kuratomi                 58
kobayashi                40
kai                      36
tom yoshio kobayashi     23
"b" area                 22
tsuda                    20
kobayashi, tom yoshio    20
rev. kai                 19
yamane, tokio            18
tom kobayashi            17
tokio yamane             16
mori                     16
george kuratomi          15
"b" area report          15
tokunaga                 15
kazama                   14
kuratomi, george         14
terada, singer           14
oshita, tomoji           14
kazama, tomio            14
shigenobu murakimi       13
yoshiyama, tom           13
isamu uchida             13
oki, kakuma              12
yamauchi, dr.            11
saito, shu               11
oscar itani              11
abe, tetsuo              11
tetsuo abe               11
nakao, masaru            11
Name: name, dtype: int64

: 

: 

As shown above, the name Kuratomi appears the most frequently among the 'riot' incident cards. Following, Kuratomi is Kobayashi, Kai, Tom Yoshio Kobayashi, Tsuda, Reverend Kai, and Tokio Yamane, to name a few. 

Also recorded in this count is "b" area or "b" area report. Area B is a significant location in Tule Lake. After the 'riot' of November 4, suspected "troublemakers" were picked up--it is recorded around 209 in November and 107 in December--and placed in "Area B" also known as the "Surveillance Area" and later designated as the "Stockade". The barracks underwent 24 hour surveillance by armed soldiers from watchtowers, were searched almost daily for "contraband", visitors were not permitted except by permission from the Tule Lake Director, and curfew was set from 7:00 P.M. to 6:00 A.M. The stockade was situated northwest of the hospital for the entire period of its existence, November 5, 1943, to August 24, 1944, and enclosed by a high wire fence. 

<center><img src="https://www.nps.gov/tule/getinvolved/images/Jail-being-built.jpg?maxwidth=1200&maxheight=1200&autorotate=false" width="400" alt="Locations"></center>
<center>*Note.* Photograph is courtesy from Densho Digital Archive, photograph, accessed from the National Park Service, Tule Lake website. [1]</center>



From the output above, we'll take 25 of the individuals listed and explore their geographical spaces in more detail; one of those being, Kuratomi.

Now that we've organized our data and identified key figures, we are prepared to begin gathering geographic information. All three datasets--the WRA, FAR, and Incident Cards-- will be used to identify different geographical spaces, specifically the WRA will be helpful for pinpointing places of origin before being forced to an assembly center as well as the location of the first assembly center, the FAR will be useful for identifying forced migration to internal and external camp locations, and the Incident Cards will be helpful for locating where individuals lived during incarceration. 

The “Japanese-American Internee Data File, 1942 – 1946”, with camp intake records of evacuated Japanese-Americans, also known as WRA Form 26 is one of the major federal records associated with the War Relocation Authority (WRA), the agency established to handle the forced relocation and detention of Japanese-Americans during World War II. At the time, clerks recorded people’s information on the WRA Form 26 when brought into assembly centers or camps, afterwards, 34 fields were encoded onto punch cards. Later on the National Archives and Records Administration (NARA) digitized these punch cards and permenantly houses these records. The WRA data files are publically available.  

<center><img src="WRA-NARA-Technical-Punchcard-Instructions.png" width="400" alt="Locations"></center>

<center><img src="WRA-NARA-Technical-Punchcard-Instructions-2.png" width="400" alt="Locations"></center>

<center><img src="WRA-NARA-Technical-Punchcard-Instructions-3.png" width="400" alt="Locations"></center>

The “Final Accountability Rosters (FAR) of Evacuees at Relocation Centers, 1944-1946, records the camp outtake of evacuees at the time of their final release or transfer. During this time, clerks recorded outtake information on individuals as they exited camps. It's important to note that some individuals exited multiple camps due to forced transfers and have more than one FAR record. The FAR records were later transcribed by Densho, Ancestry.com, among others.


<center><img src="FAR-TL-March1946-Vol1-2.png" width="900" alt="Locations"></center>

The Index Cards, also called, Incident Cards are Internal Security Case Reports” index cards, from the War Relocation Authority (Record Group 210 from 1941-47) records series. The Incident Cards reference narrative reports prepared by camp investigators, police officers, and directors of internal security, relating cases of alleged “disorderly conduct, rioting, seditious behavior,” etc. at each of the 10 camps, with detailed information on the names and addresses in the camps of the persons involved, the time and place where the alleged incident occurred, an account of what happened, and a statement of action taken by the investigating officer. There are 25,045 index cards, 63% of which (15,648) come from the Tule Lake concentration camp.  
 

<center><img src="IC-Kuratomi-Box12-Image840.png" width="500" alt="Locations"></center>

## Finding Names in the Incident Cards, WRA, & FAR

Though this data is rich with information the WRA, FAR, and Incident Cards present unique challenges for us. For example, at times it was difficult to determine the order of a persons first and last name as well as many of the names are similiar to one another with one or two characters being different. Keeping this in mind, we have to be conscientious we are looking at the correct individual(s). A method for doing this is comparing information across the three separate data files like age, residence, family, individual or alien registration number, or using details found in the Other column. At other times we found that information was incomplete, for example, the WRA has truncated last names, leaving us to deduce that we were looking at the correct person after comparing the previously mentioned fields. 

As shown previously, we will follow a similiar process by using the contains function to locate data on Kuratomi. Because I am familiar with the data, I know Kuratomi's first name is George--a good practice to get in the habit of is to become familiar with the data you are working with before manipulating and analyzing it.  

By performing a simple contains search, as shown below, on the name column we are supplied with Kuratomi's residence, including other valueable information like his age, other individuals linked to Kuratomi, case numbers, as well as notes that are recorded on that specific index card which can be helpful for identifying this is correct person. 

By looking at the table below, under New-residence we find the the location of George's barrack, 6605-E. The address can be interpreted in the following manner, 66 as the block, 05 as the ward and E as the assigned barrack. 

In [ ]:
# The contains function will return values with the element 'George Kuratomi'
incidentdf[incidentdf['name'].str.contains("george kuratomi", na=False)]

Image#   newdate casenumber offense newoffense DEATH  \
2338  Box10-0217.jpg  43-11-04   A-7-P417    riot       riot         
2366  Box10-0245.jpg  43-11-04   A-7-P139    riot       riot         
2368  Box10-0247.jpg  43-11-04    A-7-P92    riot       riot         
2369  Box10-0248.jpg  43-11-04     A-7P89    riot       riot         
5358  Box12-0840.jpg  43-11-04   A-7-P139    riot       riot         
6326  Box13-0614.jpg  43-11-04   A-7-P404    riot       riot         
6328  Box13-0616.jpg  43-11-04   A-7-P402    riot       riot         
6334  Box13-0622.jpg  43-11-04   A-7-P373    riot       riot         
6335  Box13-0623.jpg  43-11-04   A-7-P377    riot       riot         
6336  Box13-0624.jpg  43-11-04   A-7-P373    riot       riot         
6337  Box13-0625.jpg  43-11-04               riot       riot         
6338  Box13-0626.jpg  43-11-04   A-7-P375    riot       riot         
6339  Box13-0627.jpg  43-11-04   A-7-P390    riot       riot         
6376  Box13-0664.jpg  43-11-04    A-7-P89    riot       riot         
6408  Box13-0696.jpg  43-11-04   A-7-P417    riot       riot         
6417  Box13-0705.jpg  43-11-04   A-7-P139    riot       riot         
6418  Box13-0706.jpg  43-11-04    A-7-P92    riot       riot         
6421  Box13-0709.jpg  43-11-04   A-7-P286    riot       riot         

                           name  lastname firstname Other Names (known as)  \
2338            george kuratomi                                              
2366            george kuratomi                                              
2368            george kuratomi                                              
2369            george kuratomi                                              
5358         george kuratomi -1  kuratomi    george                          
6326            george kuratomi                                              
6328            george kuratomi                                              
6334            george kuratomi                                              
6335            george kuratomi                                              
6336            george kuratomi                                              
6337            george kuratomi                                              
6338            george kuratomi                                              
6339            george kuratomi                                              
6376            george kuratomi                                              
6408            george kuratomi                                              
6417            george kuratomi                                              
6418            george kuratomi                                              
6421  george kuratomi, chairman                                              

     Job/Role/Title Residence NEW-residence            X            Y  \
2338                                                                    
2366                                                                    
2368                   6605-E         66-05  635285.2772  4638923.446   
2369                                                                    
5358                                                                    
6326                                                                    
6328                                                                    
6334                                                                    
6335                                                                    
6336                                                                    
6337                                                                    
6338                                                                    
6339                                                                    
6376                                                                    
6408                                                                    
6417                            

: 

: 

The .iloc is used to view the details of the notes in the Other column. The .iloc function is an integer position based function, meaning you can use indexing for selection of a particular position or cell and row within a table (dataset or also commonly called an array). 

In [ ]:
# The below function returns information the cell with the index of 6334, 18
incidentdf.iloc[5358,18]

"TAKAHASHI\n\nMentioned during questioning of Geroge Kuratomi on 12/3/43 in connection with Kashima's funeral."

: 

: 

As with the Incident Card data, performing the contains function allows us to extract values containing Kuratomi. 

Demonstrated below, George Kuratomi, can be found in the second row of the WRA datafile. Similiar to the Incident Cards, the WRA provides personal information that can be used to confirm we are looking at the correct person. To view additional details of a row, the .iloc function can be applied here as well. In the table below we can extract the following geographic information for Kuratomi: original state, assembly center, and first camp. This data will be helpful later to find the latitute and longitude coordinates for these separate locations. 

In [ ]:
# The contains function will return results with Kuratomi.
wradf[wradf['NEW-m_lastname'].str.contains("KURATOMI", na=False)].head(5)

m_dataset                     m_pseudoid    m_camp m_lastname  \
40132  wra-master   2-poston_kuratomi_1934_akiko  2-poston   KURATOMI   
40133  wra-master  6-jerome_kuratomi_1915_george  6-jerome   KURATOMI   
40134  wra-master    6-jerome_kuratomi_1897_hide  6-jerome   KURATOMI   
40135  wra-master   6-jerome_kuratomi_1923_ikuko  6-jerome   KURATOMI   
40136  wra-master   6-jerome_kuratomi_1884_iwami  6-jerome   KURATOMI   

      NEW-m_lastname m_firstname NEW-m_firstname m_birthyear NEW-m_birthyear  \
40132       KURATOMI       AKIKO           AKIKO        1934            1934   
40133       KURATOMI      GEORGE          GEORGE        1915            1915   
40134       KURATOMI        HIDE            HIDE        1897            1897   
40135       KURATOMI       IKUKO           IKUKO        1923            1923   
40136       KURATOMI       IWAMI           IWAMI        1884            1884   

      m_gender  ...                                w_citizenshipstatus  \
40132        F  ...  Did not attend Japanese Lang. School: Does not...   
40133        M  ...  Did not attend Japanese Lang. School: Has SS #...   
40134        F  ...  Did not attend Japanese Lang. School: Has AR #...   
40135        F  ...  Did not attend Japanese Lang. School: Has neit...   
40136        M  ...  Did not attend Japanese Lang. School: Has both...   

                         w_highestgrade  \
40132  No schooling or kindergarten [J]   
40133        In U.S.: High School 4 [I]   
40134       In Japan: High School 4 [D]   
40135            In U.S.: College 1 [5]   
40136       In Japan: High School 2 [B]   

                                              w_language  \
40132            Not applicable (11 years old and under)   
40133  Japanese speak, read (& write), English speak,...   
40134      Japanese speak, read (& write), English speak   
40135   Japanese speak and English speak, read (& write)   
40136  Japanese speak, read (& write), English speak,...   

                                              w_religion w_occupqual1  \
40132                                       Buddhist [1]                
40133                                       Buddhist [1]           71   
40134                                       Buddhist [1]                
40135  No religion, undecided, none, atheist, agnosti...                
40136                                        Baptist [A]           72   

      w_occupqual2 w_occupqual3 w_occuppotn1 w_occuppotn2 w_filenumber  
40132                                                           200060  
40133          170                        27            6       956857  
40134                                    625          614       952839  
40135                                    125            4       952840  
40136           73          303          433                    953282  

[5 rows x 46 columns]

: 

: 

In [ ]:
# Apply the .iloc to a row index to view all information within that given row
wradf.iloc[40133]

m_dataset                                                       wra-master
m_pseudoid                                   6-jerome_kuratomi_1915_george
m_camp                                                            6-jerome
m_lastname                                                        KURATOMI
NEW-m_lastname                                                    KURATOMI
m_firstname                                                         GEORGE
NEW-m_firstname                                                     GEORGE
m_birthyear                                                           1915
NEW-m_birthyear                                                       1915
m_gender                                                                 M
m_originalstate                                                         CA
m_familyno                                                            4119
NEW-m_familyno                                                        4119
m_familyno_normal        

: 

: 

Lastly, we will look through the FAR to locate exit locations for George Kuratomi. Exit locations include forced movements from one camp to another throughout incarceration. 

A quick scan of the table below, we find that George Kuratomi doesn't appear among the listed names, this is because 'George' is an evangelical name used by Toshio Kuratomi. This was a phenomena discovered by simply becoming familiar with the data. In fact, George was not the only individual to use an evangelical name during incarceration. Now that we are equipped with this information, this will require us to do further investigation so we are careful not to rule out a person because their doesnt immediately appear in a dataset. It may end up being the case here where an alternative name is used.

In [ ]:
# Apply the contains function to return results for Kuratomi
fardf[fardf['NEW-last_name_corrected'].str.contains("Kuratomi", na=False)]

FAR Exit Dataset original_order far_line_id family_number  \
16101          granada1           3820        3713          4091   
16102           jerome1           3856        3811          4091   
16103           jerome1           3857        3812          4091   
16104          granada1           3819        3712          4091   
16105           jerome1           3855        3810          4091   
16106          granada1           3818        3711          4091   
16107           jerome1           3858        3813          4091   
16255           jerome1           3859        3814          4119   
16256         tulelake1          10726       10251          4119   
16257         tulelake1          10727       10252          4119   
38724           poston1           7304          41         10033   
38725           poston1           7303          40         10033   
38726           poston1           7302          39         10033   
38727           poston1           7305          42         10033   
142361          rohwer1           4581        4477        11-178   
142362          jerome1           3851        3806        11-178   
142363          rohwer1           4582        4478        11-178   
142364          jerome1           3853        3808        11-178   
142365          rohwer1           4583        4479        11-178   
142366          jerome1           3854        3809        11-178   
142367          rohwer1           4584        4480        11-178   
142368          jerome1           3852        3807        11-178   

       NEW-family_number last_name_corrected NEW-last_name_corrected  \
16101               4091            Kuratomi                Kuratomi   
16102               4091            Kuratomi                Kuratomi   
16103               4091            Kuratomi                Kuratomi   
16104               4091            Kuratomi                Kuratomi   
16105               4091            Kuratomi                Kuratomi   
16106               4091            Kuratomi                Kuratomi   
16107               4091            Kuratomi                Kuratomi   
16255               4119            Kuratomi                Kuratomi   
16256               4119            Kuratomi                Kuratomi   
16257               4119            Kuratomi                Kuratomi   
38724              10033            Kuratomi                Kuratomi   
38725              10033            Kuratomi                Kuratomi   
38726              10033            Kuratomi                Kuratomi   
38727              10033            Kuratomi                Kuratomi   
142361            11-178            Kuratomi                Kuratomi   
142362            11-178            Kuratomi                Kuratomi   
142363            11-178            Kuratomi                Kuratomi   
142364            11-178            Kuratomi                Kuratomi   
142365            11-178            Kuratomi                Kuratomi   
142366            11-178            Kuratomi                Kuratomi   
142367            11-178            Kuratomi                Kuratomi   
142368            11-178            Kuratomi                Kuratomi   

       last_name_original first_name_corrected first_name_original  ...  \
16101            Kuratomi                 Hide                Hide  ...   
16102            Kuratomi                 Hide                Hide  ...   
16103            Kuratomi                Ikuko               Ikuko  ...   
16104            Kuratomi               Masako              Masako  ...   
16105            Kuratomi               Masako              Masako  ...   
16106            Kuratomi              Rintaro             Rintaro  ...   
16107            Kuratomi              Rintaro             Rintaro  ...   
16255            Kuratomi               Toshio              Toshio  ...   
16256            Kuratomi                Utako               Utako  ...   
16257            Kuratomi          

: 

: 

In [ ]:
# Apply the .iloc function to view additional information in index (row) below
fardf.iloc[16255]

FAR Exit Dataset                     jerome1
original_order                          3859
far_line_id                             3814
family_number                           4119
NEW-family_number                       4119
last_name_corrected                 Kuratomi
NEW-last_name_corrected             Kuratomi
last_name_original                  Kuratomi
first_name_corrected                  Toshio
first_name_original                   Toshio
NEW-first_name_corrected              Toshio
other_names                           George
NEW-other_names                       George
date_of_birth                               
NEW-date_of_birth                           
year_of_birth                               
NEW-year_of_birth                           
sex                                        M
NEW_marital_status                         S
citizenship                                C
alien_registration_no.                      
type_of_original_entry                  SAAC
NEW-type_o

: 

: 

In this first module, I have shown how to use Pythons contains function to manipulate and analyze specific data within datasets. We harnessed the power of regular expressions to return different variations of values that were inconsisently entered, inlcuding limiting the contains function to return particular values. We identified key figures involved in the 'riot' at Tule Lake by using the counts value function. Futhermore, we located the place of origin, assembly center, first camp location, and place of residence at Tule Lake for one of the many persons invovled in the 'riot' that took place at Tule Lake on November 4, 1944. 

In the following module, we will focus on how to use the geographical data collected here to construct datasets that will allow us model and visualize the movements of 25 selected individuals. 

In [ ]:
# The below command let's us save the modified dataframes into a new output csv file. 
# This can be useful when using these files for further steps of processing.
caseno.to_csv('casenodf.csv', index=False)

: 

: 

### References
[1] Photograph; Tule Lake, California. Tule Lake Historic Jail Rehabilitation Project Public Review. . .; Densho Digital Archive; Found on the National Park Service, Tule Lake. [https://www.nps.gov/tule/getinvolved/tule-lake-historic-jail-rehabilitation-project-public-review.htm, May 31, 2018]

## Notebooks

The below module is organized into a sequential set of Python Notebooks that allows us to interact with the collections related to the Framework for Unlocking and Linking WWII Japanese American Incarceration Biographical Data to explore, clean, prepare, visualize and analyze it from historical context perspective.

1. A Framework for Unlocking and Linking WWII Japanese American Incarceration Biographical Data - Context Based Data Manipulation and Analysis - Part 2
2. A Framework for Unlocking and Linking WWII Japanese American Incarceration Biographical Data - Data Visualization 

: 

: 